search/v2/features_layer_path_finder.cpp

```diff
@@ -60,124 +92,103 @@ void FeaturesLayerPathFinder::FindReachableVertices(FeaturesLayerMatcher & match
   uint64_t const bottomUpCost = CalcBottomUpPassCost(layers);
 
   if (bottomUpCost < topDownCost)
-    FindReachableVerticesBottomUp(matcher, layers, reachable);
+    FindReachableVerticesBottomUp(matcher, layers, results);
   else
-    FindReachableVerticesTopDown(matcher, layers, reachable);
+    FindReachableVerticesTopDown(matcher, layers, results);
 }
 
 void FeaturesLayerPathFinder::FindReachableVerticesTopDown(
     FeaturesLayerMatcher & matcher, vector<FeaturesLayer const *> const & layers,
-    vector<uint32_t> & reachable)
+    vector<IntersectionResult> & results)
 {
-  reachable = *(layers.back()->m_sortedFeatures);
+  ASSERT(!layers.empty(), ());
```

Comment:
This method is private and is called from FindReachableVertices(), where this check is implemented.  I thought it will be redundant to check it again here.